src/mpl/configure.ac

```diff
@@ -785,6 +785,78 @@ fi
 ## END OF PTHREAD MUTEX COMMON CHECK
 #######################################################################
 
+#######################################################################
+## START OF ATOMIC CODE
+#######################################################################
+
+PAC_ARG_ATOMIC_PRIMITIVES
+
+AC_DEFUN([MPL_ATOMIC_TEST_PROGRAM], [AC_LANG_PROGRAM([[
+        #include "$1"
+    ]],[[
+        MPL_atomic_int_t a, b;
+        int c;
+
+        MPL_atomic_relaxed_store_int(&a, 0);
+        MPL_atomic_relaxed_store_int(&b, 1);
+        c = MPL_atomic_relaxed_load_int(&a);
+
+        MPL_atomic_release_store_int(&a, 0);
+        MPL_atomic_release_store_int(&b, 1);
+        c = MPL_atomic_acquire_load_int(&a);
+
+        MPL_atomic_fetch_add_int(&a, 10);
+        MPL_atomic_fetch_sub_int(&a, 10);
+
+        c = MPL_atomic_cas_int(&a, 10, 11);
+        c = MPL_atomic_swap_int(&a, 10);
+
+        MPL_atomic_write_barrier();
+        MPL_atomic_read_barrier();
+        MPL_atomic_read_write_barrier();
+    ]])]
+)
+
+dnl MPL_TRY_ATOMIC_HEADER([header file from src/ dir],[symbolic name in configure],[HAVE_ macro suffix],[feature description])
+dnl Does an AC_LINK_IFELSE() to see if the header file works.
+AC_DEFUN([MPL_TRY_ATOMIC_HEADER],[
+if test "$with_mpl_atomic_primitives" = "auto" || \
+   test "$with_mpl_atomic_primitives" = "$2"
+then
+    checked_specified_primitive=yes
+    AC_MSG_CHECKING([for support for $4])
+    SAVE_CFLAGS="$CFLAGS"
+    CFLAGS="$CFLAGS -I${srcdir}/include"
+    AC_LINK_IFELSE([MPL_ATOMIC_TEST_PROGRAM([$1])],
+        [AC_DEFINE([HAVE_$3], [1], [Define to 1 if we have support for $4])]
+        [AC_MSG_RESULT([yes])]
+    ,
+        [AC_MSG_RESULT([no])]
+    )
+    CFLAGS="$SAVE_CFLAGS"
+fi
+])
+
+AC_CHECK_SIZEOF([void *])
+
+MPL_TRY_ATOMIC_HEADER([mpl_atomic_gcc_sync.h], [gcc_sync], [GCC_INTRINSIC_SYNC], [gcc __sync intrinsics])
+MPL_TRY_ATOMIC_HEADER([mpl_atomic_gcc_atomic.h], [gcc_atomic], [GCC_INTRINSIC_ATOMIC], [gcc __atomic intrinsics])
+MPL_TRY_ATOMIC_HEADER([mpl_atomic_c11.h], [c11], [C11_ATOMICS], [C11 atomic intrinsics])
+MPL_TRY_ATOMIC_HEADER([mpl_atomic_nt_intrinsics.h], [windows], [NT_INTRINSICS], [Windows NT atomic intrinsics])
+
+if test "$with_mpl_atomic_primitives" = "auto" ||\
+   test "$with_mpl_atomic_primitives" = "lock"; then
+   AC_DEFINE(USE_LOCK_BASED_PRIMITIVES, 1, [Define to 1 if mutex-based synchronization is used])
+elif test "$with_mpl_atomic_primitives" = "unsafe" || \
+     test "$with_mpl_atomic_primitives" = "no" || \
+     test "$with_mpl_atomic_primitives" = "none"; then
+   AC_DEFINE(USE_UNSAFE_PRIMITIVES, 1, [Define to 1 if unsafe atomic primitives are used])
```

Comment:
I find `unsafe` quite ambiguous. I think just `no` is clear enough.

Reply:
I changed to make it accept only `no` and `none`.